--- a/openjdk/jdk/make/common/shared/Defs-linux.gmk	2014-03-02 09:37:39.798197699 +0200
+++ b/openjdk/jdk/make/common/shared/Defs-linux.gmk	2014-03-02 09:37:58.422316444 +0200
@@ -216,24 +216,9 @@
 
 # Macro to check it's input file for banned dependencies and verify the
 #   binary built properly. Relies on process exit code.
-ifndef CROSS_COMPILE_ARCH
-define binary_file_verification # binary_file
-( \
-  $(ECHO) "Checking for mapfile use in: $1" && \
-  if [ "`$(NM) -D -g --defined-only $1 | $(EGREP) 'SUNWprivate'`" = "" ] ; then \
-    $(ECHO) "$(MAPFILE_WARNING): File was not built with a mapfile: $1"; \
-  fi && \
-  $(ECHO) "Library loads for: $1" && \
-  $(LDD) $1 && \
-  $(ECHO) "RUNPATH for: $1" && \
-  ( $(READELF) -d $1 | $(EGREP) 'NEEDED|RUNPATH|RPATH' ) \
-)
-endef
-else
 define binary_file_verification 
 ( \
   $(ECHO) "Skipping binary file verification for cross-compile build" \
 )
 endef
-endif
 
--- a/openjdk/Makefile	2014-03-02 09:35:50.037498535 +0200
+++ b/openjdk/Makefile	2014-03-02 09:36:09.323621303 +0200
@@ -79,7 +79,7 @@
 include ./make/sponsors-rules.gmk
 include ./make/deploy-rules.gmk
 
-all:: sanity
+all::
 
 ifeq ($(SKIP_FASTDEBUG_BUILD), false)
   all:: fastdebug_build
