--- a/openjdk/jdk/make/common/shared/Defs-linux.gmk	2013-12-02 13:21:00.125451340 +0000
+++ b/openjdk/jdk/make/common/shared/Defs-linux.gmk	2013-12-02 13:21:11.585451340 +0000
@@ -112,7 +112,7 @@
 ifneq ($(ALT_OPENWIN_HOME),)
   OPENWIN_HOME :=$(call PrefixPath,$(ALT_OPENWIN_HOME))
 else
-  OPENWIN_HOME  =/usr/X11R6/
+  OPENWIN_HOME  =/opt/Xorg
 endif
 
 # DEVTOOLS_PATH: for other tools required for building (such as zip, etc.)
--- a/openjdk/jdk/make/sun/gtk/Makefile	2014-10-03 20:56:49.907928619 +0200
+++ b/openjdk/jdk/make/sun/gtk/Makefile	2014-10-03 20:56:56.505928046 +0200
@@ -54,7 +54,7 @@
 vpath %.c   $(PLATFORM_SRC)/native/common/deps/gtk2
 endif
 
-OTHER_LDLIBS = -lX11
+OTHER_LDLIBS = -L=$(OPENWIN_LIB) -lX11
 
 ifeq ($(SYSTEM_GTK), true)
   OTHER_LDLIBS += $(GTK_LIBS)
